### PR TITLE
Add file and line attributes

### PR DESCRIPTION
**Pull request by pbrisbin**
Location information isn't an official part of the JUnit XML spec, so
various conventions have emerged for reporting locations such as within
the name attribute, inferring from the classname using project structure
assumptions, or including it in the content of the testcase node (as we
do).

At least one tool* that reads JUnit reports will look for file and
line attributes if present on the testcase node. To better support this
and any similarly-behaving tools, let's add them.

\* mikepenz/action-junit-report

### CODE REVIEW

File: library/Test/Hspec/JUnit.hs [hunk 1]
```diff
@@ -108,6 +112,13 @@ itemToTestCase applyPrefix group name item = Schema.TestCase
                   )
   }
  where
+  toSchemaLocation Location {..} = Schema.Location
+    { Schema.locationFile = locationFile
+    , Schema.locationLine = if locationLine >= 0
+      then fromIntegral locationLine
+      else 0
```
Comment (eborden) [inline library/Test/Hspec/JUnit.hs]: could this be
```suggestion
    , Schema.locationLine = fromIntegral $ max 0 locationLine
```

File: library/Test/Hspec/JUnit.hs [hunk 2]
```diff
@@ -108,6 +112,13 @@ itemToTestCase applyPrefix group name item = Schema.TestCase
                   )
   }
  where
+  toSchemaLocation Location {..} = Schema.Location
```
Comment (eborden) [inline library/Test/Hspec/JUnit.hs]: Can this be top level? It isn't capturing anything from this closure, right?

**Review by eborden — COMMENTED** [review]
Are there competing implementations of `file` and `line`?

**Comment by pbrisbin** [comment]
> Are there competing implementations of file and line?

I do not know what this means, can you rephrase?

**Comment by eborden** [comment]
> > Are there competing implementations of file and line?
> 
> I do not know what this means, can you rephrase?

Are there competing standards for tracking this info in JUnit?

**Comment by pbrisbin** [comment]
> Are there competing standards for tracking this info in JUnit?

My best answer to that question is the PR description :shrug: 

I was going by this [Stack Overflow](https://stackoverflow.com/questions/4922867/what-is-the-junit-xml-format-specification-that-hudson-supports) which basically says "there is no spec", [this comment](https://github.com/mikepenz/action-junit-report/issues/40#issuecomment-981894773), which says "line numbers are not consistent between output formats", and then [this source](https://github.com/mikepenz/action-junit-report/blob/adae3952cb46310560d94874c989afa3c1b7681a/src/testParser.ts#L36-L81) which does a bunch of guess work (if `file` and `line` aren't given explicitly).

If you have any knowledge of a stronger convention than adding these two attributes explicitly do let me know. The maintainer of the GitHub Action is super nice and would be willy to look for whatever we provide -- especially if we have a standards-based reason for it.

**Review by eborden — APPROVED** [review]
Well without any competing standard, this seems like a sane enough path.

**Comment by pbrisbin** [comment]
When I implemented `JUnitConfig`, I took the option names from `jest-junit`. I don't know that it's canonical per say, but it seems mature and popular. It has a `JEST_JUNIT_ADD_FILE_ATTRIBUTE` option, but I cannot figure out what that does.

**Comment by eborden** [comment]
Looks like it adds a `file` attribute with the file path :smile: 

https://github.com/jest-community/jest-junit/commit/dee46ff7ce105a9f34502b8555563708b5bf5f22

**Comment by pbrisbin** [comment]
Ha, fair enough.
